Description: fix for schema-utils ≥ 3
Author: Xavier Guimard <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2020-11-17

--- a/lib/BannerPlugin.js
+++ b/lib/BannerPlugin.js
@@ -9,7 +9,7 @@
 const ModuleFilenameHelpers = require("./ModuleFilenameHelpers");
 const Template = require("./Template");
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/BannerPlugin.json");
 
 /** @typedef {import("../declarations/plugins/BannerPlugin").BannerPluginArgument} BannerPluginArgument */
--- a/lib/DllPlugin.js
+++ b/lib/DllPlugin.js
@@ -8,7 +8,7 @@
 const FlagAllModulesAsUsedPlugin = require("./FlagAllModulesAsUsedPlugin");
 const LibManifestPlugin = require("./LibManifestPlugin");
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/DllPlugin.json");
 
 /** @typedef {import("../declarations/plugins/DllPlugin").DllPluginOptions} DllPluginOptions */
--- a/lib/DllReferencePlugin.js
+++ b/lib/DllReferencePlugin.js
@@ -13,7 +13,7 @@
 const makePathsRelative = require("./util/identifier").makePathsRelative;
 const WebpackError = require("./WebpackError");
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/DllReferencePlugin.json");
 
 /** @typedef {import("../declarations/plugins/DllReferencePlugin").DllReferencePluginOptions} DllReferencePluginOptions */
--- a/lib/HashedModuleIdsPlugin.js
+++ b/lib/HashedModuleIdsPlugin.js
@@ -5,7 +5,7 @@
 "use strict";
 const createHash = require("./util/createHash");
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/HashedModuleIdsPlugin.json");
 
 /** @typedef {import("../declarations/plugins/HashedModuleIdsPlugin").HashedModuleIdsPluginOptions} HashedModuleIdsPluginOptions */
--- a/lib/IgnorePlugin.js
+++ b/lib/IgnorePlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/IgnorePlugin.json");
 
 /** @typedef {import("../declarations/plugins/IgnorePlugin").IgnorePluginOptions} IgnorePluginOptions */
--- a/lib/LoaderOptionsPlugin.js
+++ b/lib/LoaderOptionsPlugin.js
@@ -6,7 +6,7 @@
 
 const ModuleFilenameHelpers = require("./ModuleFilenameHelpers");
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/LoaderOptionsPlugin.json");
 
 /** @typedef {import("../declarations/plugins/LoaderOptionsPlugin").LoaderOptionsPluginOptions} LoaderOptionsPluginOptions */
--- a/lib/ProgressPlugin.js
+++ b/lib/ProgressPlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/ProgressPlugin.json");
 
 /** @typedef {import("../declarations/plugins/ProgressPlugin").ProgressPluginArgument} ProgressPluginArgument */
--- a/lib/SourceMapDevToolPlugin.js
+++ b/lib/SourceMapDevToolPlugin.js
@@ -11,7 +11,7 @@
 const createHash = require("./util/createHash");
 const { absolutify } = require("./util/identifier");
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/SourceMapDevToolPlugin.json");
 
 /** @typedef {import("../declarations/plugins/SourceMapDevToolPlugin").SourceMapDevToolPluginOptions} SourceMapDevToolPluginOptions */
--- a/lib/WatchIgnorePlugin.js
+++ b/lib/WatchIgnorePlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../schemas/plugins/WatchIgnorePlugin.json");
 
 /** @typedef {import("../declarations/plugins/WatchIgnorePlugin").WatchIgnorePluginOptions} WatchIgnorePluginOptions */
--- a/lib/debug/ProfilingPlugin.js
+++ b/lib/debug/ProfilingPlugin.js
@@ -2,7 +2,7 @@
 const path = require("path");
 const mkdirp = require("mkdirp");
 const { Tracer } = require("chrome-trace-event");
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../../schemas/plugins/debug/ProfilingPlugin.json");
 
 /** @typedef {import("../../declarations/plugins/debug/ProfilingPlugin").ProfilingPluginOptions} ProfilingPluginOptions */
--- a/lib/optimize/AggressiveSplittingPlugin.js
+++ b/lib/optimize/AggressiveSplittingPlugin.js
@@ -6,7 +6,7 @@
 
 const identifierUtils = require("../util/identifier");
 const { intersect } = require("../util/SetHelpers");
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../../schemas/plugins/optimize/AggressiveSplittingPlugin.json");
 
 /** @typedef {import("../../declarations/plugins/optimize/AggressiveSplittingPlugin").AggressiveSplittingPluginOptions} AggressiveSplittingPluginOptions */
--- a/lib/optimize/LimitChunkCountPlugin.js
+++ b/lib/optimize/LimitChunkCountPlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../../schemas/plugins/optimize/LimitChunkCountPlugin.json");
 const LazyBucketSortedSet = require("../util/LazyBucketSortedSet");
 
--- a/lib/optimize/MinChunkSizePlugin.js
+++ b/lib/optimize/MinChunkSizePlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../../schemas/plugins/optimize/MinChunkSizePlugin.json");
 
 /** @typedef {import("../../declarations/plugins/optimize/MinChunkSizePlugin").MinChunkSizePluginOptions} MinChunkSizePluginOptions */
--- a/lib/optimize/OccurrenceChunkOrderPlugin.js
+++ b/lib/optimize/OccurrenceChunkOrderPlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../../schemas/plugins/optimize/OccurrenceOrderChunkIdsPlugin.json");
 
 /** @typedef {import("../../declarations/plugins/optimize/OccurrenceOrderChunkIdsPlugin").OccurrenceOrderChunkIdsPluginOptions} OccurrenceOrderChunkIdsPluginOptions */
--- a/lib/optimize/OccurrenceModuleOrderPlugin.js
+++ b/lib/optimize/OccurrenceModuleOrderPlugin.js
@@ -4,7 +4,7 @@
 */
 "use strict";
 
-const validateOptions = require("schema-utils");
+const validateOptions = require("schema-utils").validate;
 const schema = require("../../schemas/plugins/optimize/OccurrenceOrderModuleIdsPlugin.json");
 
 /** @typedef {import("../../declarations/plugins/optimize/OccurrenceOrderModuleIdsPlugin").OccurrenceOrderModuleIdsPluginOptions} OccurrenceOrderModuleIdsPluginOptions */
