Description: fix for mkdirp ≥ 1
Author: Xavier Guimard <yadd@debian.org>
Forwarded: no
Last-Update: 2020-10-25

--- a/lib/Compiler.js
+++ b/lib/Compiler.js
@@ -469,9 +469,8 @@
 					if (targetFile.match(/\/|\\/)) {
 						const dir = path.dirname(targetFile);
 						this.outputFileSystem.mkdirp(
-							this.outputFileSystem.join(outputPath, dir),
-							writeOut
-						);
+							this.outputFileSystem.join(outputPath, dir)
+						).then(() => {writeOut()}).catch((er) => {writeOut(er)});
 					} else {
 						writeOut();
 					}
@@ -491,7 +490,7 @@
 		this.hooks.emit.callAsync(compilation, err => {
 			if (err) return callback(err);
 			outputPath = compilation.getPath(this.outputPath);
-			this.outputFileSystem.mkdirp(outputPath, emitFiles);
+			this.outputFileSystem.mkdirp(outputPath).then(() => {emitFiles()}).catch(er => {throw er});
 		});
 	}
 
@@ -517,9 +516,10 @@
 		if (!recordsOutputPathDirectory) {
 			return writeFile();
 		}
-		this.outputFileSystem.mkdirp(recordsOutputPathDirectory, err => {
-			if (err) return callback(err);
+		this.outputFileSystem.mkdirp(recordsOutputPathDirectory).then(() => {
 			writeFile();
+		}).catch( err => {
+			return callback(err);
 		});
 	}
 
--- a/lib/LibManifestPlugin.js
+++ b/lib/LibManifestPlugin.js
@@ -72,13 +72,14 @@
 							? JSON.stringify(manifest, null, 2)
 							: JSON.stringify(manifest);
 						const content = Buffer.from(manifestContent, "utf8");
-						compiler.outputFileSystem.mkdirp(path.dirname(targetPath), err => {
-							if (err) return callback(err);
+						compiler.outputFileSystem.mkdirp(path.dirname(targetPath)).then(() => {
 							compiler.outputFileSystem.writeFile(
 								targetPath,
 								content,
 								callback
 							);
+						}).catch( err => {
+							return callback(err);
 						});
 					},
 					callback
